src/jobserver.h

```diff
@@ -0,0 +1,52 @@
+// Copyright 2024 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#if defined( _WIN32)
+// windows.h defines min() and max() which conflict with std::min()
+// and std::max(), which both might be used in sources (indirectly)
+// including this header. Avoid compile errors by telling windows.h
+// to not define those two.
+#define NOMINMAX
+#include <windows.h>
+#endif
+
+#include <cstdbool>
+#include <string>
+
+#include "util.h"
+
+struct Jobserver {
+  Jobserver();
+  ~Jobserver();
+  void Init();
+  bool Enabled() const;
+  bool Acquire();
+  void Release();
+
+private:
+  bool ParseJobserverAuth(const char *type);
+  bool AcquireToken();
+  void ReleaseToken();
+
+  std::string jobserver_name_;
+  size_t token_count_;
```

Comment:
I recommend using default initialization for members here to simplify the source code, e.g.:

```
  size_t token_count_ = 0;
#ifdef _WIN32
   HANDLE sem_ = INVALID_HANDLE_VALUE;
#else
  int fd_ = -1;
#endif  
```

wdyt?

Reply:
Done